plugins/ptp_operator/event/event.go

```diff
@@ -92,42 +95,54 @@ func (dt *DependingClockState) hasMetricHelp() map[string]string {
 }
 
 // UpdateCurrentEventState ...
-func (p *PTPEventState) UpdateCurrentEventState(c ClockState) ptp.SyncState {
+func (p *PTPEventState) UpdateCurrentEventState(c ClockState, metrics map[string]*PMetric, help map[string]string) ptp.SyncState {
 	p.Lock()
 	defer func() {
 		p.Unlock()
 	}()
 	// if the current state is not locked and the new state is locked then update the current state
+
+	log.Infof("clock state %s", c.PrintClockState())
+	if _, ok2 := p.DependsOn[c.Process]; ok2 {
+		log.Infof("Has key for process  %s", c.Process)
+	} else {
+		log.Infof("Has NO key for process %s", c.Process)
+	}
+	log.Infof("depends on start %s", p.PrintDependsOn())
 	if dep, ok := p.DependsOn[c.Process]; ok {
-		index, d := dep.hasClockState(*c.IFace)
+		index, clockState := dep.hasClockState(*c.IFace)
 		// update the clock state if found or else
 		if index == -1 {
-			d = &ClockState{}
-			dep = append(dep, d)
+			clockState = &ClockState{}
+			clockState.Metric = dep.hasMetric() //possible same metrics available outside the nic stats
+			clockState.HelpText = dep.hasMetricHelp()
+			dep = append(dep, clockState)
+			p.DependsOn[c.Process] = dep
 		}
-		d.Offset = c.Offset
-		d.IFace = c.IFace
-		d.State = c.State
-		d.Process = c.Process
-		d.Value = c.Value
-		d.ClockSource = c.ClockSource
+		clockState.Offset = c.Offset
+		clockState.IFace = c.IFace
+		clockState.State = c.State
+		clockState.Process = c.Process
+		clockState.Value = c.Value
+		clockState.ClockSource = c.ClockSource
+		clockState.NodeName = c.NodeName
 		// update the metric
 		if c.Offset != nil {
-			d.Offset = pointer.Float64(*c.Offset)
+			clockState.Offset = pointer.Float64(*c.Offset)
 		}
 
 		for k, v := range c.Value {
-			iface := ""
-			if d.IFace != nil {
-				iface = *d.IFace
-			}
-			if d.Metric == nil {
-				d.Metric = dep.hasMetric()
-				d.HelpText = dep.hasMetricHelp()
+			iface := "x"
+			if clockState.IFace != nil {
+				iface = *clockState.IFace
```

Comment:
Shouldn't IFace be checked before entering the range c.Value for-loop? Otherwise if IFace is nil, then metrics with iface = 'x' will be created.

Reply:
done